cmake/G3Test.h

```diff
@@ -0,0 +1,117 @@
+#ifndef G3TEST_H_INCLUDED
```

Comment:
I think this header and associated cxx files should probably be in `core` so that other packages can use this test infrastructure.  Part of me also wants this infrastructure to be a separate PR altogether, so that it's abundantly clear that this is a new feature.

Reply:
My logic for putting it in `cmake` was that exactly so that it should be accessible to all packages, and `cmake` seemed more fundamental than even `core`. (Also, IceCube happens to keep this sort of thing in `cmake`.) However, given that all other packages depend on `core`, I think it could go there just as well. 

If you would like this as a separate PR, that would be doable; I think the main reason I didn't try to split it was that I didn't have any other immediate uses, so it seemed like a PR for the test stuff alone would not demonstrate that it really worked. I could probably invent such a minimal use, though.

Reply:
Given the way that spt3g_software is used by SO (i.e. as a separate dependency), it would be nice to allow building a test program by linking against the core library and supplying the right header.

Reply:
Yes, that is a good point. I forgot that headers for `cmake` are not installed. Unfortunately, the linking aspect is a little tricky, since the test program needs to have a `main()`, while it would be rather inappropriate for that to appear as a symbol in `libspt3g-core.so`. I'll do some brainstorming to figure out how a downstream project could utilize this. Maybe some shenanigans with hiding the actual implementation code in a macro in the header, and some kind of cmake logic to autogenerate an implmentation file into which that is expanded?

Reply:
This is now done; auto-generating an implementation file in the build directory is very slightly evil, but pretty simple.